File: coko/com.elster.jupiter.metering.cim/src/main/java/com/elster/jupiter/metering/cim/impl/AllReadingTypes.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.elster.jupiter.metering.cim.impl;

import com.elster.jupiter.metering.ReadingType;

public enum AllReadingTypes implements ReadingTypeFilter {
    INSTANCE;

    @Override
    public boolean apply(ReadingType readingType) {
        return true;
    }
}

File: coko/com.elster.jupiter.config.extjs/src/main/web/js/cfg/src/store/AllReadingTypes.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Cfg.store.AllReadingTypes', {
    extend: 'Ext.data.Store',

    fields: [
        'fullAliasName',
        'mRID'
    ],

    proxy: {
        type: 'rest',
        url: '/api/mtr/readingtypes',
        reader: {
            type: 'json',
            root: 'readingTypes'
        }
    }
});